### PR TITLE
Creation of the header page.

### PR DESCRIPTION
**Pull request by hassandogo**
image

### CODE REVIEW

File: src/app/globals.css [hunk 1]
```diff
@@ -20,6 +20,20 @@ body {
   font-family: Arial, Helvetica, sans-serif;
 }
 
+.dark {
+  --Dark-Blue: hsl(209, 23%, 22%);
+  --Very-Dark-Blue: hsl(207, 26%, 17%);
+  --Very-Dark-Blue: hsl(200, 15%, 8%);
+  --Dark-Gray: hsl(0, 0%, 52%);
```
Comment (libock-serge) [inline src/app/globals.css]: Hi @hassandogo. It's not good to have "Very Dark Blue" twice. Please rename accordingly

File: src/app/layout.tsx [hunk 2]
```diff
@@ -29,7 +30,7 @@ export default function RootLayout({
       <body
         className={`${geistSans.variable} ${geistMono.variable} antialiased`}
       >
-        <Provider>{children}</Provider>
+        {children}
```
Comment (libock-serge) [inline src/app/layout.tsx]: Please align with @duchelle5093 on the role of this Provider you are removing. We shall have a discussion tomorrow on how to handle merge conflicts. I'm on sick leave today. But general rule of thumb, avoid removing people's changes without being 100% sure you're not breaking something

**Review by libock-serge — CHANGES_REQUESTED** [review]

File: src/app/components/header.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import clsx from "clsx";
+import { AppContainer } from "../globale/container";
+import { FC } from "react";
+import { useTheme } from "next-themes";
+
+
+
+type HeaderProps = {
+  isDarkMode?: boolean;
+  // setIsDarkMode?:  Dispatch<SetStateAction<boolean>>;
```
Comment (libock-serge) [inline src/app/components/header.tsx]: Don't push commented code in a pull request

File: src/app/components/header.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          <button
+            className="flex gap-2 items-center lg:text-[1.5rem]"
+            onClick={hanldeToggleTheme}
+          >
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              width="24"
+              height="24"
+              viewBox="0 0 24 24"
+              fill={isDarkMode ? 'white' : 'white'}
```
Comment (libock-serge) [inline src/app/components/header.tsx:36]: What is the point of checking isDarkMode if the outcome is always "white" whether isDarkMode is true or false?

File: src/app/globale/spacer.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import clsx from 'clsx';
```
Comment (libock-serge) [inline src/app/globale/spacer.tsx]: Why are you adding this "spacer.tsx" in a "globale" folder?

File: src/app/globale/container.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { FC, PropsWithChildren } from 'react';
```
Comment (libock-serge) [inline src/app/globale/container.tsx]: Why a "globale" folder?

File: src/app/countriesCard/Card.tsx [hunk 7]
```diff
@@ -1,35 +1,38 @@
 import Image from "next/image"
```
Comment (libock-serge) [inline src/app/countriesCard/Card.tsx:1]: This component should go to the / app / components folder.

File: src/app/globals.css [hunk 8]
```diff
@@ -2,26 +2,73 @@
...
 
+
 @layer utilities {
   .text-balance {
     text-wrap: balance;
   }
 }
+
+
+@layer base {
```
Comment (libock-serge) [inline src/app/globals.css:17]: Be careful with copy-paste. Are you using all the styles below?

File: src/app/layout.tsx [hunk 9]
```diff
@@ -24,12 +27,23 @@ export default function RootLayout({
...
+
+
   return (
     <html lang="en">
       <body
-        className={`${geistSans.variable} ${geistMono.variable} antialiased`}
+        className={`${geistSans.variable} ${geistMono.variable} antialiased bg-secondary text-secondary-foreground`}
       >
-        <Provider>{children}</Provider>
+        <ThemeProvider
```
Comment (libock-serge) [inline src/app/layout.tsx:38]: What is this provider that you are removing from the root?

**Review by libock-serge — CHANGES_REQUESTED** [review]

File: src/app/components/header.tsx (hunk 4 above)
Comment (hassandogo) [reply src/app/components/header.tsx:36]: Sa permet a "moon-outline" de ne pas changer son background

File: src/app/globale/spacer.tsx (hunk 5 above)
Comment (hassandogo) [reply src/app/globale/spacer.tsx]: j'ai mit juste pour configurer les space mais j'ai déjà enlevé.

File: src/app/countriesCard/Card.tsx (hunk 7 above)
Comment (hassandogo) [reply src/app/countriesCard/Card.tsx:1]: D'accord

File: src/app/globals.css (hunk 8 above)
Comment (hassandogo) [reply src/app/globals.css:17]: Okay d'accord.

File: src/app/globale/container.tsx (hunk 6 above)
Comment (hassandogo) [reply src/app/globale/container.tsx]: j'ai supprimer et j'ai déplacer son composant container.tsx dans le /app/component.

**Comment by hassandogo** [comment]
Voici la partie header avec son "dark mode" :
Capture d'écran 2024-10-03 155144
Capture d'écran 2024-10-03 155241

File: package.json [hunk 10]
```diff
@@ -9,12 +9,21 @@
...
+    "@tanstack/react-query-next-experimental": "^5.59.0",
+    "class-variance-authority": "^0.7.0",
+    "clsx": "^2.1.1",
+    "ionicons": "^7.4.0",
+    "lucide-react": "^0.447.0",
     "next": "14.2.13",
+    "next-themes": "^0.3.0",
     "react": "^18",
     "react-dom": "^18",
+    "shadcn-ui": "^0.9.2",
```
Comment (libock-serge) [inline package.json:24]: Why are you adding shadcn to the project?

**Review by libock-serge — APPROVED** [review]

**Pull request merged**